dev/com.ibm.ws.beanvalidation.v20.cdi/src/com/ibm/ws/beanvalidation/v20/cdi/internal/LibertyHibernateValidatorExtension.java

```diff
@@ -0,0 +1,183 @@
+           immediate = true,
+           property = { "api.classes=" +
+                        "javax.validation.Validator;" +
+                        "javax.validation.ValidatorFactory;" +
+                        "org.hibernate.validator.HibernateValidatorFactory;" +
+                        "org.hibernate.validator.cdi.internal.ValidatorFactoryBean;" +
+                        "org.hibernate.validator.cdi.HibernateValidator;" +
+                        "org.hibernate.validator.cdi.ValidationExtension;" +
+                        "org.hibernate.validator.cdi.internal.interceptor.ValidationInterceptor;" +
+                        "org.hibernate.validator.internal.engine.ValidatorImpl;" +
+                        "org.hibernate.validator.internal.engine.ValidatorFactoryImpl;" +
+                        "org.hibernate.validator.cdi.internal.InjectingConstraintValidatorFactory;" +
+                        "com.ibm.ws.beanvalidation.v20.cdi.internal.LibertyValidatorBean;" +
+                        "com.ibm.ws.beanvalidation.v20.cdi.internal.LibertyValidatorFactoryBean",
+                        "service.vendor=IBM"
+           })
+public class LibertyHibernateValidatorExtension implements Extension, WebSphereCDIExtension {
+    private static final TraceComponent tc = Tr.register(LibertyHibernateValidatorExtension.class);
+
+    private static final PrivilegedAction<ThreadContextAccessor> getThreadContextAccessorAction = new PrivilegedAction<ThreadContextAccessor>() {
+        @Override
+        public ThreadContextAccessor run() {
+            return ThreadContextAccessor.getThreadContextAccessor();
+        }
+    };
+
+    /**
+     * Privileged action for creating a Validation20ClassLoader.
+     */
+    private class CreateValidation20ClassLoaderAction implements PrivilegedAction<Validation20ClassLoader> {
+        private final ClassLoader parentCL;
+        private final String moduleHint;
+
+        private CreateValidation20ClassLoaderAction(ClassLoader parentCL, String moduleHint) {
+            this.parentCL = parentCL;
+            this.moduleHint = moduleHint;
+        }
+
+        @Override
+        public Validation20ClassLoader run() {
+            return new Validation20ClassLoader(parentCL, moduleHint);
+        }
+    }
+
+    private ValidatorFactoryBean vfBean;
+    private ValidatorFactoryBean hibernateValidatorFactoryBean;
+    private ValidatorBean vBean;
+    private ValidatorBean hibernateValidatorBean;
+
+    public void afterBeanDiscovery(@Observes AfterBeanDiscovery afterBeanDiscoveryEvent, BeanManager beanManager) {
+        if (vfBean == null) {
+            vfBean = hibernateValidatorFactoryBean = new LibertyValidatorFactoryBean();
+            afterBeanDiscoveryEvent.addBean(vfBean);
+            afterBeanDiscoveryEvent.addBean(hibernateValidatorFactoryBean);
+        }
+
+        if (vBean == null) {
+            vBean = hibernateValidatorBean = new LibertyValidatorBean();
```

Comment:
Similar to above, why do we have separate fields for vBean and hibernateValidatorBean when they are always the same value, and why do we add it twice?